docs/handbook/components/component.mdx

```diff
@@ -0,0 +1,192 @@
+        let output = match dest {
+            // For on-chain submissions, the output is abi encoded using the encode_trigger_output function from trigger.rs
+            Destination::Ethereum => Some(encode_trigger_output(trigger_id, &res)),
+            // For local testing via wasi-exec, the output is returned as raw bytes
+            Destination::CliOutput => Some(WasmResponse {
+                payload: res.into(),
+                ordering: None
+            }),
+        };
+        Ok(output)
+    }
+}
+```
+
+Components can contain any compatible logic, including [blockchain interactions](./blockchain-interactions), [network requests](./network-requests) , off-chain computations, and more. To learn about the types of components that WAVS is best suited for, visit the [design considerations](../../design) page.
+
+#### Logging in a component
+
+Components can use logging to debug and track the execution of the component.
+
+**Logging in development**:
+
+Use `println!()` to write to stdout/stderr. This is visible when running `make wasi-exec` locally in the template.
+
+```rust lib.rs
+println!("Debug message: {:?}", data);
+```
+
+**Logging in production**
+
+For production, you can use a `host::log()` function which takes a `LogLevel` and writes its output via the tracing mechanism. Along with the string that the developer provides, it attaches additional context such as the `ServiceID`, `WorkflowID`, and component `Digest`.
+
+```rust lib.rs
+use bindings::host::{self, LogLevel};
+
+host::log(LogLevel::Info, "Production logging message");
+```
+
+### Component output
+
+After processing data in the `run` function, the component can encode the output data for submission back to Ethereum. In the template example, this is done using the `encode_trigger_output` function in the trigger.rs file.
+
+```rust
+/// Encodes the output data for submission back to Ethereum
+pub fn encode_trigger_output(trigger_id: u64, output: impl AsRef<[u8]>) -> WasmResponse {
+    WasmResponse {
+        payload: solidity::DataWithId {
+            triggerId: trigger_id,
+            data: output.as_ref().to_vec().into(),
+        }
+        .abi_encode(),  // ABI encode the struct for blockchain submission
+        ordering: None, // Optional ordering parameter for transaction sequencing
+    }
+}
+```
+
+Outputs for components are returned as a `WasmResponse` struct, which is a wrapper around the output data of the component for encoding and submission back to Ethereum. It contains a `payload` field that is the encoded output data and an optional `ordering` field that is used to order the transactions in the workflow. The `WasmResponse` is submitted to WAVS which routes it according to the workflow's submission logic.
+
+## Component definition
+
+A component is defined in the [workflow](../workflows) object of the [service.json](../service) file. Below is an example of the different fields that can be defined in the component object.
+
+```json service.json
```

Comment:
It does feel like we define component sources many times. We should link out maybe if we do 

service.mdx, workflows.mdx, and component.mdx. or we give just the very specific sections for each at each. I think 1 large service json file would be better utilized